src/tzdata/build.jl

```diff
@@ -39,21 +39,19 @@ function build(
     # TODO: Deprecate skipping tzdata installation step or use `nothing` instead
     if !isempty(tz_source_dir)
         @info "Installing $version tzdata region data"
-        regions = union!(intersect(regions, readdir(artifact_dir)), CUSTOM_REGIONS)
-        for region in setdiff(regions, CUSTOM_REGIONS)
-            cp(joinpath(artifact_dir, region), joinpath(tz_source_dir, region), force=true)
-        end
-        # Copy over our 'custom regions' from `deps/custom_tzsource_regions`
-        custom_tz_source_dir = joinpath(dirname(dirname(@__DIR__)), "deps", "custom_tzsource_regions")
-        for region in CUSTOM_REGIONS
-            cp(joinpath(custom_tz_source_dir, region), joinpath(tz_source_dir, region), force=true)
+        region_paths = vcat(
+            joinpath.(artifact_dir, intersect!(readdir(artifact_dir), regions)),
+            joinpath.(CUSTOM_REGION_DIR, CUSTOM_REGIONS),
+        )
+        for path in region_paths
+            cp(path, joinpath(tz_source_dir, basename(path)), force=true)
         end
     end
 
     # TODO: Deprecate skipping conversion step or use `nothing` instead
     if !isempty(compiled_dir)
         @info "Converting tz source files into TimeZone data"
-        tz_source = TZSource(joinpath.(tz_source_dir, regions))
+        tz_source = TZSource(readdir(tz_source_dir, join=true))
```

Comment:
There are some nuisances to the original code here I'm not sure I still agree with but I'll make some adjustments to keep the original behaviour. I'll be sure to add a detailed comment on the subtleties